diff --git a/SpellChecker.json.in b/SpellChecker.json.in
index c71badad3db1..1eeaf998c1c5 100644
--- a/SpellChecker.json.in
+++ b/SpellChecker.json.in
@@ -1,8 +1,10 @@
 {
+    "Id": "spellchecker",
     "Name" : "SpellChecker",
     "Version" : "3.7.0",
     "CompatVersion" : "3.7.0",
     "Vendor" : "Carel Combrink",
+    "VendorId" : "carelcombrink",
     "Copyright" : "(C) 2015 - 2024 Carel Combrink",
     "License" : [ "GNU Lesser General Public License Usage",
                   "",
