commit da19594f7e2e8804612b59db83f5eef282fe451c
Author: Eugene Shalygin <eugene.shalygin@gmail.com>
Date:   Thu Jun 24 18:22:52 2021 +0200

    Fix build with Clang 12 and install the native lib

diff --git a/sources/libClangSharp/CMakeLists.txt b/sources/libClangSharp/CMakeLists.txt
index c91c5f9..fdeb444 100644
--- a/sources/libClangSharp/CMakeLists.txt
+++ b/sources/libClangSharp/CMakeLists.txt
@@ -18,12 +18,6 @@ set(SOURCES
   CXType.h
 )
 
-set(LIBS
-  clangAST
-  clangFrontend
-  libclang
-)
-
 set(LLVM_SEARCH_PATHS
   ${PATH_TO_LLVM}
   ${PATH_TO_LLVM}/lib/cmake
@@ -37,12 +31,19 @@ find_package(Clang REQUIRED CONFIG
              PATHS ${LLVM_SEARCH_PATHS}
              NO_DEFAULT_PATH)
 
-if(WIN32)
-  add_library(libClangSharp SHARED ${SOURCES})
-  target_include_directories(libClangSharp PUBLIC ${LLVM_INCLUDE_DIRS})
-  target_link_libraries(libClangSharp ${LIBS})
-else()
-  add_library(ClangSharp SHARED ${SOURCES})
-  target_include_directories(ClangSharp PUBLIC ${LLVM_INCLUDE_DIRS})
-  target_link_libraries(ClangSharp ${LIBS})
-endif()
+add_library(ClangSharp SHARED ${SOURCES})
+target_include_directories(ClangSharp PRIVATE ${CLANG_INCLUDE_DIRS})
+target_link_libraries(ClangSharp PRIVATE clang-cpp libclang)
+set_target_properties(ClangSharp PROPERTIES
+    PREFIX lib
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION}
+)
+
+include(GNUInstallDirs)
+
+install(TARGETS ClangSharp
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT runtime
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT development
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime
+)
