diff --git a/sources/libClangSharp/ClangSharp.cpp b/sources/libClangSharp/ClangSharp.cpp
index 9c8ef35..fbc3d32 100644
--- a/sources/libClangSharp/ClangSharp.cpp
+++ b/sources/libClangSharp/ClangSharp.cpp
@@ -3898,7 +3898,7 @@ CXString clangsharp_Cursor_getStringLiteralValue(CXCursor C) {
         const Stmt* S = getCursorStmt(C);
 
         if (const StringLiteral* SL = dyn_cast<StringLiteral>(S)) {
-            return createDup(SL->getString());
+            return createDup(SL->getBytes());
         }
     }
 
